gov/politeia/proposals.go

```diff
@@ -90,371 +87,509 @@ func NewProposalsDB(politeiaURL, dbPath string) (*ProposalDB, error) {
+	}
 
-		// increment since the save is successful.
-		proposalsSaved++
+	// Update vote results data on finished proposals that are not yet
+	// fully synced with politeia.
+	err = db.proposalsVoteResultsUpdate()
+	if err != nil {
+		return err
 	}
 
-	return proposalsSaved, nil
+	log.Info("Politeia records were synced.")
+
+	return nil
 }
 
-// AllProposals fetches all the proposals data saved to the db.
-func (db *ProposalDB) AllProposals(offset, rowsCount int,
-	filterByVoteStatus ...int) (proposals []*pitypes.ProposalInfo,
-	totalCount int, err error) {
+// ProposalsAll fetches the proposals data from the local db.
+// The argument filterByVoteStatus is optional.
+//
+// Satisfies the PoliteiaBackend interface.
+func (db *ProposalsDB) ProposalsAll(offset, rowsCount int,
+	filterByVoteStatus ...int) ([]*pitypes.ProposalRecord, int, error) {
+	// Sanity check
 	if db == nil || db.dbP == nil {
 		return nil, 0, errDef
 	}
 
 	var query storm.Query
+
 	if len(filterByVoteStatus) > 0 {
-		// Filter by the votes status
 		query = db.dbP.Select(q.Eq("VoteStatus",
-			pitypes.VoteStatusType(filterByVoteStatus[0])))
+			ticketvotev1.VoteStatusT(filterByVoteStatus[0])))
 	} else {
 		query = db.dbP.Select()
 	}
 
 	// Count the proposals based on the query created above.
-	totalCount, err = query.Count(&pitypes.ProposalInfo{})
+	totalCount, err := query.Count(&pitypes.ProposalRecord{})
 	if err != nil {
-		return
+		return nil, 0, err
 	}
 
 	// Return the proposals listing starting with the newest.
+	var proposals []*pitypes.ProposalRecord
 	err = query.Skip(offset).Limit(rowsCount).Reverse().OrderBy("Timestamp").
 		Find(&proposals)
-	if err != nil && err != storm.ErrNotFound {
+	if err != nil && !errors.Is(err, storm.ErrNotFound) {
 		log.Errorf("Failed to fetch data from Proposals DB: %v", err)
-	} else {
-		err = nil
 	}
 
-	return
+	return proposals, totalCount, nil
 }
 
-// ProposalByToken returns the single proposal identified by the provided token.
-func (db *ProposalDB) ProposalByToken(proposalToken string) (*pitypes.ProposalInfo, error) {
+// ProposalByToken retrieves the proposal for the given token argument.
+//
+// Satisfies the PoliteiaBackend interface.
+func (db *ProposalsDB) ProposalByToken(token string) (*pitypes.ProposalRecord, error) {
 	if db == nil || db.dbP == nil {
 		return nil, errDef
 	}
 
-	return db.proposal("TokenVal", proposalToken)
+	return db.proposal("Token", token)
 }
 
-// ProposalByRefID returns the single proposal identified by the provided refID.
-// RefID is generated from the proposal name and used as the descriptive part of
-// the URL to proposal details page on the /proposal page.
-func (db *ProposalDB) ProposalByRefID(RefID string) (*pitypes.ProposalInfo, error) {
-	if db == nil || db.dbP == nil {
-		return nil, errDef
+// fetchVettedTokens fetches all vetted tokens ordered by the timestamp of
+// their last status change.
+func (db *ProposalsDB) fetchVettedTokensInventory() ([]string, error) {
+	page := 0
```

Comment:
Do you mean to have a `page++` at the end of this loop?  It looks like it would iterate forever on the same page if the number of tokens were `recordsv1.InventoryPageSize`.
Otherwise, `page` is not a useful var.

Reply:
omg yes, sorry about that, my bad